consensus/src/consensus_observer/observer/consensus_observer.rs

```diff
@@ -63,13 +67,13 @@ pub struct ConsensusObserver {
     active_observer_state: ActiveObserverState,
 
     // The block payload store (containing the block transaction payloads)
-    block_payload_store: BlockPayloadStore,
+    block_payload_store: Arc<Mutex<BlockPayloadStore>>,
 
     // The ordered block store (containing ordered blocks that are ready for execution)
-    ordered_block_store: OrderedBlockStore,
+    ordered_block_store: Arc<Mutex<OrderedBlockStore>>,
 
     // The pending block store (containing pending blocks that are without payloads)
-    pending_block_store: PendingBlockStore,
+    pending_block_store: Arc<Mutex<PendingBlockStore>>,
```

Comment:
why this needs to be Arc of Mutex? this is not going to the callback right?

Reply:
Yeah, you're right, it's not technically needed now. I just wanted to keep it consistent (i.e., make all blocks stores thread safe), so I don't need to worry if something changes in the future 😄